sdk/python/lib/test/langhost/marshal_failure/__main__.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2016-2018, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pulumi import CustomResource, Output, ResourceOptions
+from pulumi.runtime import invoke
+
+class MyResource(CustomResource):
+    value: Output[str]
+
+    def __init__(self, name, value, opts):
+        CustomResource.__init__(self, "test:index:MyResource", name, props={
+            "value": value,
+        }, opts=opts)
+
+invoke("test:index:MyFunction", props={})
+
+resA = MyResource("resourceA", "foo", None)
+
+resB = MyResource("resourceB", "bar".encode("utf8"), ResourceOptions(depends_on=[resA]))
```

Comment:
For my own edification, it is this line here that causes the undhandled exception during marshaling, because `.encode()` returns a `bytes` which we do not know how to marshal, correct?

Reply:
Correct.